pkg/storage/formattedstore.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package storage
+
+import (
+	"fmt"
+)
+
+const (
+	failOpenUnderlyingStore      = "failed to open underlying store: %w"
+	failCloseUnderlyingStore     = "failed to close underlying store: %w"
+	failCloseAllUnderlyingStores = "failed to close all underlying stores: %w"
+
+	failFormatKeyValuePair      = "failed to format key-value pair: %w"
+	failPutInUnderlyingStore    = "failed to put encrypted document in underlying store: %w"
+	failGetFromUnderlyingStore  = "failed to get formatted value from underlying store: %w"
+	failParseFormattedValue     = "failed to parse formatted value: %w"
+	failDeleteInUnderlyingStore = "failed to delete key-value pair in underlying store: %w"
+)
+
+// Formatter represents a type that can encrypt and decrypt between
+// Structured Documents and Encrypted Documents.
+type Formatter interface {
+	FormatPair(k string, v []byte) ([]byte, error)
+	ParseValue([]byte) ([]byte, error)
+}
+
+// FormattedProvider is a storage provider that allows for data to be formatted in an underlying provider.
+type FormattedProvider struct {
```

Comment:
I think the fact that we are creating encrypted/confidential stores could imply a stronger terminology than `Formatted` and FormattedProvider.

Reply:
I renamed the concrete EDV Formatter to EncryptedFormatter. However, the Formatter interface and the FormattedProvider type are general enough to be easily used for something that isn't encryption, which is why I didn't give them a more specific name. It just so happens that using it for an encrypted format is a concrete use case that I'm adding.